Arrowgene.Ddon.GameServer/Handler/CraftStartEquipGradeUpHandler.cs

```diff
@@ -0,0 +1,190 @@
+            uint currentTotalEquipPoint = equipItem.EquipPoints;
+
+            List<CDataCommonU32> gradeuplist = new() { new CDataCommonU32(gearupgradeID) };
+            CDataEquipSlot EquipmentSlot = new();
+            CDataCurrentEquipInfo CurrentEquipInfo = new()
+            {
+                ItemUId = equipItemUID,
+                EquipSlot = EquipmentSlot
+            };
+
+            // More dummy data, looks like its DragonAugment related.
+            CDataCraftStartEquipGradeUpUnk0Unk0 DragonAugmentData = new();
+            CDataCraftStartEquipGradeUpUnk0 dummydata = new() // TODO: Figure this out
+            {
+                Unk0 = new List<CDataCraftStartEquipGradeUpUnk0Unk0> { DragonAugmentData },
+                DragonAugment = false,    // makes the DragonAugment slot popup appear if set to true.
+            };
+
+            var res = new S2CCraftStartEquipGradeUpRes();
+            S2CItemUpdateCharacterItemNtc updateCharacterItemNtc = new();
+
+            // Update item equip points in the database
+            uint addEquipPoint = (uint)((IsGreatSuccess ? 300 : 180) * (0.8 + (Random.Shared.NextDouble() * 0.4))); 
+            currentTotalEquipPoint += addEquipPoint;
+            Server.Database.UpdateItemEquipPoints(equipItemUID, currentTotalEquipPoint);
+
+            // Removes crafting materials
+            foreach (var craftMaterial in request.CraftMaterialList)
+            {
+                var updateResults = _itemManager.ConsumeItemByUIdFromMultipleStorages(Server, client.Character, ItemManager.BothStorageTypes, craftMaterial.ItemUId, craftMaterial.ItemNum);
+                updateCharacterItemNtc.UpdateItemList.AddRange(updateResults);
+            }
+
+            // Subtract less Gold if support pawn is used and add slightly more points
+            if (request.CraftSupportPawnIDList.Count > 0)
+            {
+                goldRequired = (uint)(goldRequired * 0.95);
+                currentTotalEquipPoint += 10;
+            }
+
+            var updateWalletPoint = Server.WalletManager.RemoveFromWallet(client.Character, WalletType.Gold, goldRequired);
+            updateCharacterItemNtc.UpdateWalletList.Add(updateWalletPoint);
+
+            // Handling the check on how many points we need to upgrade the weapon in its current state.
+            ClientItemInfo itemInfo = ClientItemInfo.GetInfoForItemId(Server.AssetRepository.ClientItemInfos, equipItem.ItemId);
+            byte currentStars = (byte)itemInfo.Quality;
+            int requiredPoints = currentStars switch
+            {
+                0 => 350,
+                1 => 700,
+                2 => 1000,
+                3 => 1500,
+                4 => 800,  // TODO: Check if the gear can reach "True" level (4) and handle that properly with "canContinue"
+                _ => throw new InvalidOperationException("Invalid star level")
+            };
+
+            if (currentStars == 3) canContinue = false;
```

Comment:
Use braces on this if statement instead of doing same line. This type of code is prone to bugs.

Do either 
```c#
if (currentStars == 3)
{
    canContinue = false
}
```

or 

```c#
canContinue = (currentStars < 3);
```